### PR TITLE
Pla 472 hailstorm add eslintrc

### PR DESCRIPTION
**Pull request by Coderwelsch**
This MR adds tooling for linting and format checking to scripts

### CODE REVIEW

File: .prettierrc.cjs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    trailingComma: 'es5',
+    tabWidth: 4,
+    semi: true,
+    singleQuote: true,
```
Comment (rodelta) [inline .prettierrc.cjs:5]: Do we have a established format for this? 

I like the standard of double quotes (and would simplify to read the diff of this PR), but if there's already a pattern, we can go for this. 
Prettier has pretty good defaults, and IMHO when we go for alternatives it should be well documented.

File: .github/workflows/format-check.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: "npm"
+      - run: npm ci
+      - run: npm run format:check
```
Comment (rodelta) [inline .github/workflows/format-check.yml:22]: To save on pipeline time, I would do this in the lint job. They're both pretty fast, and both have to do with code structure.

**Review by rodelta — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -22,12 +22,16 @@
     "url": "https://github.com/abusix/hailstorm/issues"
   },
   "scripts": {
-    "rollup": "NODE_ENV=production rollup -c",
     "storybook": "storybook dev -p 6006",
     "clean": "rimraf ./dist",
     "type-check": "tsc --project tsconfig.json",
     "test": "jest",
-    "build": "npm run clean && npm run rollup",
+    "lint:check": "eslint src --ext .ts,.tsx",
```
[the remark this answers is not in the document]
Comment (Coderwelsch) [reply package.json]: Done ☺️☺️

File: .eslintrc.cjs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        'airbnb-typescript',
+        'airbnb/hooks',
+        'plugin:react/recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:storybook/recommended',
+        'plugin:prettier/recommended',
+    ],
+    plugins: ['react', '@typescript-eslint', 'prettier'],
+    rules: {
+        'import/prefer-default-export': 'off',
```
[the remark this answers is not in the document]
Comment (Coderwelsch) [reply .eslintrc.cjs:23]: Good point! I replaced the numbers with their corresponding string literals :)

File: .github/workflows/format-check.yml (hunk 2 above)
Comment (Coderwelsch) [reply .github/workflows/format-check.yml:22]: I was thinking of it as well! Ok then I will put them together ☺️

Comment (Coderwelsch) [reply .github/workflows/format-check.yml]: Ok, done!

**Review by rodelta — APPROVED** [review]
Approving to not delay this, since it didn't get an extra review

**Pull request merged**